### PR TITLE
Ignore docs and examples from the published npm package

### PR DESCRIPTION
**Pull request by brettimus**
*Issue #, if available:* Closes https://github.com/amzn/style-dictionary/issues/914

*Description of changes:*

This PR changes the `.npmignore` to exclude files related to examples and documentation from the published NPM package. 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

**Review by tripu — APPROVED** [review]
Static analysis tools looking for vulnerabilities are finding a file `style-dictionary/examples/advanced/custom-file-header/node_modules/json5/package.json` and screaming [CVE-2022-46175](https://nvd.nist.gov/vuln/detail/CVE-2022-46175).

Yet another reason to exclude examples from the npm tarball.

So, +10 from my and my colleagues.

**Review by coliff — APPROVED** [review]

**Review by dbanksdesign — APPROVED** [review]
LGTM! Thank you for this!

**Comment by chazzmoney** [comment]
Unfortunately, the SD executable expects the examples folder to be present. However, you could remove / hide the advanced folder. Or, you could suggest a PR to remove the init functionality.

**Comment by dbanksdesign** [comment]
Ah good catch @chazzmoney! You are right, this would break `style-dictionary init` .

**Comment by chazzmoney** [comment]
For clarity, this is the code that would be broken:

https://github.com/amzn/style-dictionary/blob/main/bin/style-dictionary#L60-L67

so if you retain examples/basic and examples/complete and npmignore examples/advanced, it would remove that vulnerability.

Or, maybe we could be convinced to remove the init...

File: .npmignore [hunk 1]
```diff
@@ -3,3 +3,6 @@
 .editorconfig
 images/
 test/
+examples/
```
Comment (chazzmoney) [inline .npmignore:6]: See the comments on the PR. Can't exclude this folder directly because it breaks functionality. Other options would work though.

**Review by chazzmoney — CHANGES_REQUESTED** [review]

**Comment by jorenbroekema** [comment]
Revamping docs, examples in the near future, will try to ensure if we still put them on NPM, they at least don't have security issues, but will close this particular PR as it would break init command

**Comment by brettimus** [comment]
Sorry I never followed up on this PR! But yeah, I think for teams who need to be meticulous about every dependency and sub-dependency they use, having minimal deps for the examples would be much appreciated. Will be happy to see #914 closed 🙏

**Pull request closed without merging**